pkg/backend/httpstate/client/api.go

```diff
@@ -280,7 +280,7 @@ func pulumiAPICall(ctx context.Context,
 	// backwards compatibility.
 	req.Header.Set("User-Agent", UserAgent())
 	// Specify the specific API version we accept.
-	req.Header.Set("Accept", "application/vnd.pulumi+8")
+	req.Header.Add("Accept", "application/vnd.pulumi+8")
```

Comment:
Hrm this is now not gonna be correct I think.  This will mean that for the template download step we accept both `application/x-tar` and `application/vnd.pulumi+8` responses.  But in reality we only accept the former.  I think if `Accept` is already in `opts.Header` we should not add this at all. (And I'd revert this and below to `.Set`)

Reply:
The service will reject any request that doesn't contain `application/vnd.pulumi+${n}` where `${n} > 3`. Without a change in the service that is otherwise unrelated to this feature, I think `Accept: application/vnd.pulumi+8 application/x-tar` is the best we will get.

The alternative is to completely give up on content negotiation and just specify an API version.